xpra/server/server_core.py

```diff
@@ -1901,7 +1903,7 @@ def auth_failed(msg):
         if not proto.authenticators:
             socktype = conn.socktype_wrapped
             try:
-                proto.authenticators = self.make_authenticators(socktype, remote, conn)
+                proto.authenticators = self.make_authenticators(socktype, remote, conn, getattr(proto, 'http_request', None))
```

Comment:
Using `getattr` here is a bit ugly.
Why not stash the `http_request` object in the connection's options instead?
Then the `make_authenticators` function doesn't need any changes and the authenticator can just extract the `http_request` from the connection?

Reply:
> Why not stash the `http_request` object in the connection's options instead?

No good reason.  Mainly my unfamiliarity with this code.  Will `conn` allways be a sub-class of `xpra.net.bytestreams.Connection`?  Are there any namespace consideration to avoid colliding with some other piece of code?

It looks like you have dropped python 2.x support.  Are you now accepting addition of `typing` annotations?  I think a few strategically placed annotations could make this code considerably easier to follow.

Reply:
`conn` will always either be a subclass or use the same structure, as for `options` although collisions are technically possible and not actively prevented, there is no sane reason for re-using the `http_request` name for something else.

We have already added a lot of `typing` annotations since python<3.6 was dropped, adding more would be most welcome.

Reply:
> ... there is no sane reason for re-using the http_request name for something else.

I should have phrased my question differently.  I am not certain who has control over key names in the `conn.options` dict.  I think your reply implied that only xpra code does.  (as opposed to server configuration, or perhaps even remote clients)

I'll switch to using `conn.options['http_request']`.